internal/engine/upper_test.go

```diff
@@ -2238,11 +2315,11 @@ func TestUpperBuildImmediatelyAfterCrashRebuild(t *testing.T) {
 		assert.Equal(t, podbuilder.FakeContainerIDSet(1), ms.LiveUpdatedContainerIDs)
 	})
 
-	// Restart the pod with a new container id, to simulate a container restart.
-	f.podEvent(podbuilder.New(t, manifest).
-		WithPodID("pod-id").
-		WithContainerID("funnyContainerID").
-		Build(), manifest.Name)
+	// Simulate a container restart where the new container isn't running yet
+	// HACK(milas): by making the container ID an empty string, it'll get ignored; we should really just
+	// 	properly set the container status
+	pb = pb.WithContainerID("")
```

Comment:
AFAICT the old logic here was actually subtly wrong - we want the same Pod, but a restarted container.

(The test actually started failing because our logic kicked in and pruned the other pod 😬)